packages/node-core/src/indexer/blockDispatcher/block-dispatcher.ts

```diff
@@ -87,27 +84,23 @@ export abstract class BlockDispatcher<B, DS>
     this.processQueue.abort();
   }
 
-  enqueueBlocks(heightsBlocks: B[] | number[], latestBufferHeight?: number): void {
+  enqueueBlocks(heightsBlocks: (IBlock<B> | number)[], latestBufferHeight: number): void {
     // In the case where factors of batchSize is equal to bypassBlock or when heights is []
     // to ensure block is bypassed, we set the latestBufferHeight to the heights
     // make sure lastProcessedHeight in metadata is updated
     if (!!latestBufferHeight && !heightsBlocks.length) {
       heightsBlocks = [latestBufferHeight];
```

Comment:
I wonder if there's a better way to do this. I notice that when dictionary results are empty this results in fetching the block which takes some time.

The current reason we do this is so that metadata is updated but not before other blocks in the queue are indexed